### ISSUE TITLE
Trying to vendor fork

### ISSUE THREAD

**Issue by neelance** [kardianos/govendor#121]
I'm trying to vendor a fork:
```
govendor fetch gopkg.in/gorp.v1::github.com/sourcegraph/gorp@v1
```
I don't get the expected result of branch `v1` of `github.com/sourcegraph/gorp` being vendored into `vendor/gopkg.in/gorp.v1`.

**Comment by kardianos** [comment]
It's a regression. :/
Right now I don't have tests for sync and fetch. I'm working on adding them now and the framework needed for them. I'll have this fixed before v1 is released.

**Comment by kardianos** [comment]
There is definitely a bug in that it isn't putting it in the correct folder. That's the regression.

There might also be a limitation of design.
Right now govendor is choosing *tag* v1.7.1 over *branch* v1. Right now `govendor` will only choose the branch over the tag if they are exactly the same. And because "v1.7.1" is assumed to be higher than "v1" (branch), then it chooses "v1.7.1". 

Go (lang) source chooses to prefix branches differently then tags, which would make this a non-issue. I'm not sure how to resolve this ambiguity. Ideas?

**Comment by kardianos** [comment]
And there was a flaw in my existing test that would have warned me of this.
Fixing now.

**Comment by sean-** [comment]
GitHub - hashicorp/go-version: A Go (golang) library for parsing and verifying versions and version constraints.

https://github.com/hashicorp/go-version

If it's helpful to you. -sc

-- 
Sean Chittenden

**Comment by kardianos** [comment]
@sean- 
Oh, I see what you mean. Because no inequality was specified, you would expect goversion to find the exact match, that is the branch named "v1". Whereas govendor currently is designed to have an implicit ">=" to each tag, avoiding version ranges altogether.

@neelance If you had specified "...@>=v1" then you would have expected it to find a larger version. Otherwise you expect it to match the tag or branch exactly, correct?

I'm open to supporting inequalities in version specifiers. I would strongly like to support prefixes, such as "ssh-v1" (esp for tags) or "release-branch-v2" (or similar). 

In rust's cargo, they imply the "^" operator if nothing else is specified: http://doc.crates.io/crates-io.html#using-cratesio-based-crates . If using the parlance of "go-version", I (think) I would want version constraints specified as "=" or "~>" (or something similar). It makes no sense to me to use an operator that allows changing a major version. And allowing upper constraints such as "<" mean the package author has broken semver, which I'm strongly against allowing for; it is a bug and must not be worked around; it must be fixed.

I would appreciate thoughts on this subject. Thanks.

**Comment by kardianos** [comment]
I've added an exact version operator "=" prefix.

I'm hesitant to add plain inequalities such as ">" or "<" as I don't want to allow bumping major versions in any way. I also don't like the idea of the less-than operator.

I'm very much open to thoughts on this subject. Thanks.

**Comment by sean-** [comment]
Agreed, only equality matters. As a developer, I want the stability of a version, but if something drifts, I also want to know that I have to chase the new version at some point.  Without an eye on possible security updates, keeping track of versions of vendored code is a PITA.  When tracking a branch, I would want to stay on `v1` `master`.  But, when `v2` shows up, I'd want to be notified that `v2` exists so I could switch branches on my schedule.  Operators wouldn't help in these situations.  If `v2` came out and I didn't want `v2`, an `ignore: ["v2","=v1.7.1"]` attribute would be ideal.  Staying aware of what is current is part of the responsibility of maintaining vendored packages.

**Comment by kardianos** [comment]
I like the idea of searching for newer versions like "v2" when you are on "v1" greatly. I've filed #122 to track.

@neelance 
I think the command `govendor fetch gopkg.in/gorp.v1::github.com/sourcegraph/gorp@=v1` should do what you are seeking.
